traffic_portal/app/src/common/modules/form/topology/edit/FormEditTopologyController.js

```diff
@@ -37,11 +37,12 @@ var FormEditTopologyController = function(topologies, cacheGroups, $scope, $cont
 		saveLabel: 'Update'
 	};
 
-	$scope.save = function(name, description, topologyTree) {
-		let normalizedTopology = topologyUtils.getNormalizedTopology(name, description, topologyTree);
-		topologyService.updateTopology(normalizedTopology).
+	$scope.save = function(currentName, newName, description, topologyTree) {
+		let normalizedTopology = topologyUtils.getNormalizedTopology(newName, description, topologyTree);
+		topologyService.updateTopology(normalizedTopology, currentName).
 			then(function(result) {
-				messageModel.setMessages(result.data.alerts, false);
+				messageModel.setMessages(result.data.alerts, true);
+				locationUtils.navigateToPath('/topologies');
```

Comment:
Why navigate to `/topologies`? Don't we want to stay on the same page?

Reply:
We can't since the `query param` in the url link will have old param. So, we need to go back to /topologies, and then click on a topology, to update the `query param` to new name.

Reply:
I'm not much of a javascripter, but is it possible to use the `result.data.response.name` to navigate back to the original page?

Reply:
I am not sure if that would work but I can try.

Reply:
Updated.